File: src/test/java/com/twsg/lnl/kotlin/c5_delegate/map/g1_java/GradeTest.java
package com.twsg.lnl.kotlin.c5_delegate.map.g1_java;

import org.junit.Test;

import java.util.HashMap;
import java.util.Map;

import static java.lang.Double.doubleToLongBits;
import static org.junit.Assert.assertEquals;

/**
 * @author tonyl
 */
public class GradeTest {

    @Test
    public void get_grade_should_return_value_from_map() {
        Map<String, Double> gradeMap = new HashMap<String, Double>() {{
            put("math", 5D);
            put("physics", 4.5D);
        }};

        Grade grade = new Grade(gradeMap);

        assertEquals(doubleToLongBits(5D), doubleToLongBits(grade.getMath()));
        assertEquals(doubleToLongBits(4.5D), doubleToLongBits(grade.getPhysics()));
    }

    @Test
    public void set_grade_should_store_value_to_map() {
        Map<String, Double> gradeMap = new HashMap<>();

        Grade grade = new Grade(gradeMap);
        grade.setMath(2.5D);
        grade.setPhysics(4D);

        assertEquals(doubleToLongBits(2.5D), doubleToLongBits(gradeMap.get("math")));
        assertEquals(doubleToLongBits(4D), doubleToLongBits(gradeMap.get("physics")));
    }
}

File: src/test/java/com/twsg/lnl/kotlin/c5_delegate/map/g2_kotlin/GradeTest.kt
package com.twsg.lnl.kotlin.c5_delegate.map.g2_kotlin

import org.junit.Assert.assertEquals
import org.junit.Test
import java.lang.Double.doubleToLongBits

/**
 * @author tonyl
 */
class GradeTest {

    @Test
    fun `get grade should return value from_map`() {
        val gradeMap = mutableMapOf(
                "math" to 5.0,
                "physics" to 4.5
        )

        val grade = Grade(gradeMap)

        assertEquals(doubleToLongBits(5.0), doubleToLongBits(grade.math))
        assertEquals(doubleToLongBits(4.5), doubleToLongBits(grade.physics))
    }

    @Test
    fun `set grade should store value to map`() {
        val gradeMap = mutableMapOf<String, Double>()

        val grade = Grade(gradeMap)
        grade.math = 3.0
        grade.physics = 2.0

        assertEquals(doubleToLongBits(3.0), doubleToLongBits(gradeMap["math"]!!))
        assertEquals(doubleToLongBits(2.0), doubleToLongBits(gradeMap["physics"]!!))
    }
}